specification/sql/resource-manager/Microsoft.Sql/preview/2020-02-02-preview/DevOpsAudit.json

```diff
@@ -0,0 +1,397 @@
+        "tags": [
+          "DevOpsAudit"
+        ],
+        "description": "Lists DevOps audit settings of a server.",
+        "operationId": "ServerDevOpsAuditPolicies_ListByServer",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved server DevOps audit settings.",
+            "schema": {
+              "$ref": "#/definitions/ServerDevOpsAuditingPolicyListResult"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 404 ServerNotInSubscriptionResourceGroup - Specified server does not exist in the specified resource group and subscription.\n\n * 404 SubscriptionDoesNotHaveServer - The requested server was not found"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List DevOps audit settings of a server": {
+            "$ref": "./examples/ServerDevOpsAuditSettingsList.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ServerDevOpsAuditPolicyProperties": {
+      "description": "Properties of a server DevOps audit policy.",
+      "required": [
+        "state"
+      ],
+      "type": "object",
+      "properties": {
+        "isAzureMonitorTargetEnabled": {
+          "description": "Specifies whether DevOps audit events are sent to Azure Monitor. \r\nIn order to send the events to Azure Monitor, specify 'State' as 'Enabled' and 'IsAzureMonitorTargetEnabled' as true.\r\n\r\nWhen using REST API to configure DevOps audit, Diagnostic Settings with 'DevOpsOperationsAudit' diagnostic logs category on the master database should be also created.\r\n\r\nDiagnostic Settings URI format:\r\nPUT https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Sql/servers/{serverName}/databases/master/providers/microsoft.insights/diagnosticSettings/{settingsName}?api-version=2017-05-01-preview\r\n\r\nFor more information, see [Diagnostic Settings REST API](https://go.microsoft.com/fwlink/?linkid=2033207)\r\nor [Diagnostic Settings PowerShell](https://go.microsoft.com/fwlink/?linkid=2033043)\r\n",
+          "type": "boolean"
+        },
+        "state": {
+          "description": "Specifies the state of the policy. If state is Enabled, storageEndpoint or isAzureMonitorTargetEnabled are required.",
+          "enum": [
+            "Enabled",
+            "Disabled"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "BlobAuditingPolicyState",
+            "modelAsString": false
```

Comment:
Why is modelAsString false?

Reply:
The state is a boolean (Enabled/Disabled).

Also, This enum is shared for auditingSettings API ( regular audit) and this new API (devopsAuditingSettings).
this was reused and its actually (along with other properties) is shared for both APIs.
I am afraid changing this may break the auditingSettings API.

Reply:
What's odd to me here is that the type is a "string", yet modelAsString is false, how does the autorest generator handle that? does it come out as a bool? an int?

Reply:
hope i understood your question correctly, it comes as Enum ( where string mapped to enum val) :

https://msdata.visualstudio.com/Database%20Systems/_git/DsMainDev?path=%2FSql%2Fxdb%2Fmanifest%2Fsvc%2Fmgmt%2FWebApiHosting%2FARM%2FClients%2FDotNet%2FV20200202Preview%2FGenerated%2FModels%2FBlobAuditingPolicyState.cs&_a=contents&version=GBmaster